Sources/StreamChatUI/ImageCollectionViewCell.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ImageCollectionViewCell: _CollectionViewCell, UIScrollViewDelegate {
+    open class var reuseId: String { String(describing: self) }
+    
+    open var content: ChatMessageDefaultAttachment! {
+        didSet { updateContentIfNeeded() }
+    }
+    
+    open var imageSingleTapped: (() -> Void)?
+    
+    public private(set) lazy var imageView = UIImageView()
+        .withoutAutoresizingMaskConstraints
+    
+    public private(set) lazy var imageScrollView = UIScrollView()
+        .withoutAutoresizingMaskConstraints
+    
+    override public func defaultAppearance() {
+        super.defaultAppearance()
+        
+        imageView.clipsToBounds = true
+        imageView.contentMode = .scaleAspectFit
+    }
+    
+    override open func setUp() {
+        super.setUp()
+        
+        imageScrollView.delegate = self
+        imageScrollView.minimumZoomScale = 1
+        imageScrollView.maximumZoomScale = 5
+        
+        let doubleTapGestureRecognizer = UITapGestureRecognizer(
+            target: self,
+            action: #selector(imageScrollViewDoubleTapped)
+        )
+        doubleTapGestureRecognizer.numberOfTapsRequired = 2
+        imageScrollView.addGestureRecognizer(doubleTapGestureRecognizer)
+        
+        let singleTapGestureRecognizer = UITapGestureRecognizer(
+            target: self,
+            action: #selector(imageScrollViewSingleTapped)
+        )
+        singleTapGestureRecognizer.numberOfTapsRequired = 1
+        singleTapGestureRecognizer.require(toFail: doubleTapGestureRecognizer)
+        imageScrollView.addGestureRecognizer(singleTapGestureRecognizer)
+    }
+    
+    @objc
+    open func imageScrollViewDoubleTapped() {
+        if imageScrollView.zoomScale != imageScrollView.minimumZoomScale {
+            imageScrollView.setZoomScale(imageScrollView.minimumZoomScale, animated: true)
+        } else {
+            imageScrollView.setZoomScale(imageScrollView.maximumZoomScale / 2, animated: true)
+        }
+    }
+    
+    @objc
+    open func imageScrollViewSingleTapped() {
+        imageSingleTapped?()
+    }
+    
+    override open func setUpLayout() {
+        super.setUpLayout()
+        
+        contentView.embed(imageScrollView)
+        imageScrollView.embed(imageView)
+        imageView.pin(anchors: [.height, .width], to: contentView)
+    }
+    
+    override open func updateContent() {
+        super.updateContent()
+
+        imageView.loadImage(from: content.imageURL, resizeAutomatically: false)
```

Comment:
do we want to load the preview image url (if available first) - and only then the full image? Also should I try to reuse the image already loaded from the message list?